lib/archive_validator/collection_validator.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+class ArchiveValidator
+  class CollectionValidator < BaseValidator
+    def validate
+      collection.keep_if do |item|
```

Comment:
This seems like an appropriate place for a collection map operation. I had never seen keep_if before either. It's probably doing the same thing just more esoteric.  If we want to use the map maybe something maybe like:
```ruby
messages = collection.map { |item| 
        subvalidator = entity_validator.new(archive_hash, item)
        messages.concat(subvalidator.messages) if subvalidator.valid?
}.compact
`

Reply:
`keep_if` actually changes `collection`, so the `map` wouldn't work.  Also I'm fine with the `keep_if`, but I first didn't expect a `validate` method actually changing the collection, but with the comment above the method (which wasn't there before) I think that's OK now.

Reply:
> keep_if actually changes collection, so the map wouldn't work.

Is his code changing the collection?  That didn't jump out at me.  It looks like he's populating the messages array from inside.  My code is wrong because I did the concat inside and instead the second line should have been `subvalidator.messages if subvalidator.valid?` though.  I'm fine either way.  I was addressing your readability comment with my suggestion but if you are fine with it then so be it.

Reply:
Yes, it wasn't obvious for me either when I read it the first time, but that was without the comment that is now above the method. That now explains what's going on here and that it changes the `collection`.

Reply:
> Yes, it wasn't obvious for me either when I read it the first time, but that was without the comment that is now above the method. That now explains what's going on here and that it changes the collection.

Is there any way to structure the code to make it more self documenting?  That'd be beyond the one method obviously.  The juice may not be worth the squeeze but I figured it couldn't hurt to ask.